### PR TITLE
feat(runtime): option for setting terminal open by default

### PR DESCRIPTION
**Pull request by AriPerkkio**
- Closes https://github.com/stackblitz/tutorialkit/issues/132
- Example usage https://github.com/AriPerkkio/tutorial-vite-plugin/pull/19
- Adds also test cases that won't be run before https://github.com/stackblitz/tutorialkit/pull/241 is merged

Adds new `terminal.open` option that can be used to set terminal open by default for specific lesson.

```yaml
---
type: lesson
title: Default
terminal:
  open: true       # <-- Terminal will be open when page is navigated to
  panels: terminal
---
```

### CODE REVIEW

File: test/ui/test/terminal.test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  await expect(page.getByRole('heading', { level: 1, name: 'Terminal test - Default' })).toBeVisible();
+
+  const tab = page.getByRole('tab', { name: 'Terminal' });
+  const panel = page.getByRole('tabpanel', { name: 'Terminal' });
+
+  /* eslint-disable multiline-comment-style */
+  // TODO: Requires #245
+  // await expect(tab).not.toBeVisible();
+  // await expect(panel).not.toBeVisible();
```
Comment (AriPerkkio) [inline test/ui/test/terminal.test.ts:16]: https://github.com/stackblitz/tutorialkit/issues/245

File: packages/components/react/src/core/Terminal/index.tsx [hunk 2]
```diff
@@ -13,13 +13,28 @@ export interface TerminalRef {
 export interface TerminalProps {
   theme: 'dark' | 'light';
   className?: string;
+  id?: string;
+  role?: ComponentProps<'div'>['role'];
+  'aria-labelledby'?: string;
```
Comment (Nemikolh) [inline packages/components/react/src/core/Terminal/index.tsx:18]: Oh I'm starting to think we should forward everything:

```tsx
export interface TerminalProps extends ComponentProps<'div'> {
  theme: 'dark' | 'light';
  readonly?: boolean;
  onTerminalReady?: (terminal: XTerm) => void;
  onTerminalResize?: (cols: number, rows: number) => void;
}

...

export const Terminal = forwardRef<TerminalRef, TerminalProps>(
  ({ theme, readonly = true, onTerminalReady, onTerminalResize, ...divProps }, ref) => {

...

    return <div {...divProps} ref={divRef} />;
```

What do you think?

Comment (AriPerkkio) [reply packages/components/react/src/core/Terminal/index.tsx]: Sure, done!

**Review by Nemikolh — APPROVED** [review]
Good work on this!! :tada:

**Pull request merged**